output/cloud/output_test.go

```diff
@@ -449,30 +398,30 @@ func testCloudOutputStopSendingMetric(t *testing.T, stopOnError bool) {
 
 	count := 1
 	max := 5
-	tb.Mux.HandleFunc(fmt.Sprintf("/v1/metrics/%s", out.referenceID),
-		func(w http.ResponseWriter, r *http.Request) {
-			count++
-			if count == max {
-				type payload struct {
-					Error cloudapi.ErrorResponse `json:"error"`
-				}
-				res := &payload{}
-				res.Error = cloudapi.ErrorResponse{Code: 4}
-				w.Header().Set("Content-Type", "application/json")
-				data, err := json.Marshal(res)
-				if err != nil {
-					t.Fatal(err)
-				}
-				w.WriteHeader(http.StatusForbidden)
-				_, _ = w.Write(data)
-				return
+	tb.Mux.HandleFunc("/v1/metrics/12", func(w http.ResponseWriter, r *http.Request) {
+		count++
+		if count == max {
+			type payload struct {
+				Error cloudapi.ErrorResponse `json:"error"`
 			}
-			body, err := io.ReadAll(r.Body)
-			assert.NoError(t, err)
-			receivedSamples := []Sample{}
-			assert.NoError(t, json.Unmarshal(body, &receivedSamples))
-		})
+			res := &payload{}
+			res.Error = cloudapi.ErrorResponse{Code: 4}
```

Comment:
If it's possible, let's maybe change this magic number `4` to anything more meaningful, if not it's okay since it's just formating

Reply:
I don't know, I see we have this specific logic here: https://github.com/grafana/k6/blob/9e3359b7dd4b6dec7440d9e2cac6948b3ed64e4a/output/cloud/output.go#L582-L584

@na-- do you now more about it?

Reply:
IIRC, this is a specific error that the backend return when they won't accept any more metrics. So, when k6 sees that, the cloud output just stops prematurely. The actual test execution continues, since for local `k6 run` tests the end-of-test summary (or any other outputs) will still work, but the `cloud` output doesn't send any more metrics. And, from what I can see, `K6_CLOUD_STOP_ON_ERROR` is enabled, the cloud output will stop not just itself, but the whole test run too.

Reply:
Added a comment with mostly what Ned said. Check c2efd34771314ec1235c18ec8be163f8eae9d8b1